addons/slack/install.js

```diff
@@ -13,64 +13,94 @@ const getTemplateFiles = (fileNames) =>
   }, {});
 
 module.exports = async (ctx) => {
-  let storageSpecification = await Sdk.createStorage(ctx);
+  if (ctx.caller.permissions) {
+    // Create the Add-On Handler
 
-  // Create the Add-On Handler
-  try {
-    await Sdk.createFunction(ctx, {
-      configurationSerialized: `# Add-on configuration settings
+    // Capture resource path and remove it as we will not need it in the addon instance
+    const eventsResourcePathKey = 'fusebit_events_resource_path';
+    let eventsResourcePath = ctx.body.configuration[eventsResourcePathKey];
+    delete ctx.body.configuration[eventsResourcePathKey];
+
+    await Sdk.createFunction(
+      ctx,
+      {
+        configurationSerialized: `# Add-on configuration settings
 ${Object.keys(ctx.body.configuration)
   .sort()
   .map((k) => `${k}=${ctx.body.configuration[k]}`)
   .join('\n')}
-
-  # Storage configuration settings
-${Object.keys(storageSpecification)
-  .sort()
-  .map((k) => `${k}=${storageSpecification[k]}`)
-  .join('\n')}
 `,
-      nodejs: {
-        files: getTemplateFiles(['index.js', 'package.json', 'slack.js', 'health.js']),
-      },
-      metadata: {
-        fusebit: {
-          editor: {
-            navigationPanel: {
-              hideFiles: ['index.js'],
+        nodejs: {
+          files: getTemplateFiles(['index.js', 'package.json', 'slack.js', 'health.js']),
+        },
+
+        security: {
+          // External calls from customer infrastructure
+          // Internal calls from connector instance with Events API messages
+          authentication: 'optional',
+          authorization: [
+            {
+              action: 'function:execute',
+              resource:
+                '/account/{{accountId}}/subscription/{{subscriptionId}}/boundary/{{boundaryId}}/function/{{functionId}}/',
             },
+          ],
+          functionPermissions: {
+            allow: [
+              ...(eventsResourcePath
```

Comment:
Pull this logic out to ln 23 so that all of the eventsResourcePath stuff happens there, and this just has the `allow: [ ...eventsResourcePermissions ]`.  That'll also give you a place to add a comment or two describing why it's optional.

Reply:
I added a comment near the line with an explanation, but it seems clearer to keep the block there so you can easily see the function permissions at a glance instead of having to scroll up in the file.